enumerables.rb

```
@@ -0,0 +1,169 @@
+          temp_array << item if yield(item)
+        end
+        temp_array
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_all?
+
+  def my_all?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == true
+
+        return false
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_any?
+
+  def my_any?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == true || temp_array.nil?
+      end
+    else
+      true
+    end
+  end
+
+  # my_none?
+
+  def my_none?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == false || temp_array.nil?
+      end
+    else
+      true
+    end
+  end
+
+  # my_count
+
+  def my_count
+    if instance_of?(Array)
+      temp_array = []
+      my_each do |item|
+        temp_array << item if yield(item)
+      end
+      temp_array.size
+    end
+    return unless block_given?
+  end
+
+  # my_map
+
+  # def my_map(&block)
+  #    if block_given?
+  #         temp_array = []
+  #         self.my_each do |item|
+  #             temp_array << block.call(item)
+  #         end
+  #         temp_array
+  #         else
+  #         self.to_enum
+  #     end
+  # end
+
+  # my_inject
+
+  def my_inject(param = nil)
+    if block_given?
+      if param.nil?
+        result = shift
+        my_each_with_index do |_, index|
+          result = yield(result, self[index])
+        end
+        return result
+      else
+        result = 0
+        i = param
+        (i...size).each do |n|
+          result = yield(result, self[n])
+        end
+      end
+    else
+      puts 'No block given'
```

No debugging information should be pushed to the repo. Please remove it. You can raise an error if need be.